markup.go

```diff
@@ -160,6 +162,35 @@ func Class(class ...string) Applyer {
 	})
 }
 
+// mustValidateClassNames ensures no class names have spaces
+// and panics with clear instructions on how to fix this user error.
+func mustValidateClassNames(class []string) {
+	for _, name := range class {
+		if strings.Contains(name, " ") {
```

Comment:
It would be nice to not have to pull in the `strings` dependency. Could you replace this with: 

```
for _, name := range class {
    for _, c := range name {
        if c == ' ' {
            panic("vecty: invalid class string with spaces passed to vecty.Class")
        }
    }
}
```

instead?

Reply:
That leads to pretty hard to read code and questions like "why not simplify this by using `strings.Contains`" for readers down the line.

What about creating a helper func and documenting it:

```Go
// contains reports whether substr is within s.
// It's a copy of strings.Contains to avoid importing all of strings for size reasons.
func contains(s, substr string) bool { ... }
```

Reply:
Note that we do not need full substring containment here; i.e. we're more asking here "does the string contain the byte ' '"? rather than an entire substring, which is more complex to handle.

So we could write that in the form of a helper function:

```Go
// contains tells if the byte b is in the given string s.
//
// It's equivalent to strings.Contains(s, []byte(substr)[0]) but has a simpler
// implementation and doesn't require importing the entire strings package
// (for size reasons).
func contains(s string, b byte) bool {
    for _, c := range s {
        if c == b {
            return true
        }
    }
    return false
}
```

Our other code then becomes:

```Go
for _, name := range class {
    if contains(name, ' ') {
        panic(`vecty: invalid class string "` + name + `" with spaces passed to vecty.Class`)
    }
}
```

But I'm not really sure that is any more clear than just adding a comment:

```diff
for _, name := range class {
+   // Note: we use a for loop here instead of strings.Contains to avoid pulling in the
+   // entire strings package for size reasons.
    for _, c := range name {
        if c == ' ' {
            panic(`vecty: invalid class string "` + name + `" with spaces passed to vecty.Class`)
        }
    }
}
```

WDYT @shurcooL ? which of these is better? I'm not really sure

Reply:
My preference would be to extract it to a function.

Reply:
@slimsag Sorry, I made a mistake and wrote the wrong thing halfway through my reply. What I meant to suggest was:

```Go
// containsSpace reports whether s contains a space character.
func containsSpace(s string) bool {
    for _, c := range []byte(name) {
        if c == ' ' {
            return true
        }
    }
    return false
}
```

I didn't mean to suggest reimplementing all of `strings.Contains`, that'd be silly and unneeded.

Reply:
i'm happy either way. pending @shurcooL 's opinion and i'll update my pr.

thanks for the quick response everyone :v:

Reply:
@shurcooL ops, did not refresh my page before i submitted the comment lol. I like the `containsSpace` function. Although i don't think we need to convert name into a byte slice, since ranging over a string returns a slice of runs and `' '` is a rune character. i'll add that function :)

Reply:
(I chose to range over bytes rather than runes simply because I figured it might be faster (it may not actually be), but you're welcome to range over runes if you prefer that.)

Reply:
Makes sense -- thanks for the insight @shurcooL !

I think ranging over bytes makes more sense, but I'm OK with either way.

Reply:
@slimsag I compared the compiled versions of both functions and the []byte one does an extra copy and the bundle size is 20 lines more. Unnecessarily nitty, I know haha.

Reply:
Awesome, thanks for doing that!